public/app/features/playlist/PlaylistPage.tsx

```diff
@@ -1,32 +1,43 @@
 import React, { FC, useState } from 'react';
 import { connect, MapStateToProps } from 'react-redux';
-import { NavModel, SelectableValue, urlUtil } from '@grafana/data';
+import { NavModel } from '@grafana/data';
 import Page from 'app/core/components/Page/Page';
 import { StoreState } from 'app/types';
 import { GrafanaRouteComponentProps } from '../../core/navigation/types';
 import { getNavModel } from 'app/core/selectors/navModel';
 import { useAsync } from 'react-use';
-import { getBackendSrv, locationService } from '@grafana/runtime';
 import { PlaylistDTO } from './types';
-import { Button, Card, Checkbox, Field, LinkButton, Modal, RadioButtonGroup, VerticalGroup } from '@grafana/ui';
-import { contextSrv } from 'app/core/core';
+import { Button, Card, ConfirmModal, LinkButton } from '@grafana/ui';
+import { contextSrv } from 'app/core/services/context_srv';
 import PageActionBar from 'app/core/components/PageActionBar/PageActionBar';
 import EmptyListCTA from '../../core/components/EmptyListCTA/EmptyListCTA';
+import { deletePlaylist, getAllPlaylist } from './api';
+import { StartModal } from './StartModal';
 
 interface ConnectedProps {
   navModel: NavModel;
 }
 
-interface Props extends ConnectedProps, GrafanaRouteComponentProps {}
+export interface PlaylistPageProps extends ConnectedProps, GrafanaRouteComponentProps {}
 
-export const PlaylistPage: FC<Props> = ({ navModel }) => {
+export const PlaylistPage: FC<PlaylistPageProps> = ({ navModel }) => {
   const [searchQuery, setSearchQuery] = useState('');
   const [startPlaylist, setStartPlaylist] = useState<PlaylistDTO | undefined>();
+  const [playlistToDelete, setPlaylistToDelete] = useState<PlaylistDTO | undefined>();
+  const [forcePlaylistsFetch, setForcePlaylistsFetch] = useState(0);
 
   const { value: playlists, loading } = useAsync(async () => {
-    return getBackendSrv().get('/api/playlists', { query: searchQuery }) as Promise<PlaylistDTO[]>;
-  });
+    return getAllPlaylist(searchQuery) as Promise<PlaylistDTO[]>;
+  }, [forcePlaylistsFetch]);
   const hasPlaylists = playlists && playlists.length > 0;
+  const onDismissDelete = () => setPlaylistToDelete(undefined);
+  const onDeletePlaylist = () => {
+    playlistToDelete &&
+      deletePlaylist(playlistToDelete?.id).finally(() => {
+        setForcePlaylistsFetch(forcePlaylistsFetch + 1);
+        setPlaylistToDelete(undefined);
+      });
+  };
```

Comment:
Maybe rewrite this as 
```suggestion
  const onDeletePlaylist = () => {
    if(!playlistToDelete) {
     return;
    }
      deletePlaylist(playlistToDelete.id).finally(() => {
        setForcePlaylistsFetch(forcePlaylistsFetch + 1);
        setPlaylistToDelete(undefined);
      });
  };
```